Modules/Microsoft365DSC/Modules/M365DSCUtil.psm1

```diff
@@ -3482,6 +3490,60 @@ function New-M365DSCMissingResourcesExample
     }
 }
 
+
+
+
+<#
+.Description
+This function validates there are no updates to the module or it's dependencies and no multiple versions are present on the local system.
+
+.Parameter Force
+Specifies that all dependencies should be forcefully imported again.
+
+.Example
+Test-M365DSCModuleValidity
+
+.Example
+Test-M365DSCModuleValidity -Force
+
+.Functionality
+Public
+#>
+function Test-M365DSCModuleValidity
+{
+    [CmdletBinding()]
+    param(
+        [parameter()]
+        [Switch]
+        $Force
+    )
+    $InformationPreference = 'Continue'
+
+    # validate only one installation of the module is present (and it's the latest version available from the psgallery)
+    $latestVersion = (Find-Module -Name 'Microsoft365DSC').Version
+    Install-Module -Name 'Microsoft365DSC' -Force -AllowClobber -RequiredVersion $latestVersion
```

Comment:
I would not install a new version of M365DSC within a method called Test- … it would be sufficient to check for newer versions of m365dsc within psgallery. All dependency updates do require a new release of m365dsc.

Reply:
i absolutely agree, just forgot to remove it.